apps/src/componentLibrary/button/Button.tsx

```diff
@@ -1,36 +1,250 @@
+  icon,
+  size = 'm',
+  type = 'primary',
+  color = 'purple',
+  buttonType = 'button',
+  /** <a> specific props */
+  useAsLink = false,
+  href,
+  target,
+  download,
+  title,
+  /** <button> specific props */
+  onClick,
+  value,
+  name,
+}) => {
+  const ButtonTag = useAsLink ? 'a' : 'button';
+
+  const tagSpecificProps =
+    ButtonTag === 'a'
+      ? {
+          href: disabled ? undefined : href,
+          target,
+          /** Copied from old button component. Only need it for the older browsers,
+           *  since modern browsers (~2020+ release year secures these vulnerabilities by default) */
+          // Opening links in new tabs with 'target=_blank' is inherently insecure. Unfortunately, we depend
+          // on this functionality in a couple of place. Fortunately, it is possible to partially mitigate some of
+          // the insecurity of this functionality by using the `rel` tag to block some of the potential exploits.
+          // Therefore, we do so here.
+          rel: target === '_blank' ? 'noopener noreferrer' : undefined,
+          download,
+          title,
+        }
+      : {type: buttonType, onClick, value, name};
+
+  // Check if correct props combination is passed
+  useMemo(
+    () =>
+      checkButtonPropsForErrors({
+        type,
+        icon,
+        useAsLink,
+        buttonType,
+        onClick,
+        href,
+        download,
+      }),
+    [type, icon, useAsLink, buttonType, onClick, href, download]
+  );
+
+  /** Handling isPending state content & spinner show logic here.
+     - If there's only text - we show only spinner.
+     - If there's only icon - we show only spinner.
+     - If there's text and iconLeft or both iconLeft and iconRight -> we show spinner on the left + text + iconRight (if it's present).
+     - If there's text and iconRight - we show text + spinner on the right.
+     */
+  const showIcon = icon && !isPending;
+  const showIconLeft = iconLeft && !isPending;
+  const showIconRight =
+    (iconRight && !isPending) || (isPending && iconRight && iconLeft);
+  const addPendingButtonWithHiddenTextClass =
+    isPending && !icon && !iconLeft && !iconRight;
+  const spinnerPosition = iconRight && !iconLeft ? 'right' : 'left';
+
+  return (
+    <ButtonTag
+      className={classNames(
+        moduleStyles.button,
+        moduleStyles[`button-${type}`],
+        moduleStyles[`button-${color}`],
+        moduleStyles[`button-${size}`],
+        addPendingButtonWithHiddenTextClass &&
+          moduleStyles.buttonPendingWithHiddenText,
+        className
+      )}
+      id={id}
+      disabled={disabled}
+      aria-disabled={disabled}
+      aria-label={ariaLabel}
+      {...tagSpecificProps}
+    >
+      {isPending && spinnerPosition === 'left' && (
+        <FontAwesomeV6Icon {...spinnerIcon} />
+      )}
+      {showIconLeft && <FontAwesomeV6Icon {...iconLeft} />}
+      {showIcon && <FontAwesomeV6Icon {...icon} />}
+      {text && <span>{text}</span>}
```

Comment:
Out of curiosity - why are we using CSS to hide the text span when showing the loading spinner, but using conditionals to hide the icons?

Reply:
It's for the case when we have only text in the button  - for the pending state we want to hide the text and only show spinner. We hide text with css in order to keep button sizes the same when we hide the text (if we don't render text as hidden for pending state - button size will be changed)

For render/not render icons - size of the buttons will remain the same cause pending icon will simply replace the not rendered icon